common/lib/capa/capa/safe_exec/tests/test_safe_exec.py

```diff
@@ -94,6 +98,81 @@ def test_can_do_something_forbidden_if_run_unsafely(self):
         self.assertEqual(g['files'], os.listdir('/'))
 
 
+class TestLimitConfiguration(unittest.TestCase):
+    """
+    Test that resource limits can be configured and overriden via Django settings.
+
+    We just test that the limits passed to `codejail` as we expect them to be.
+    Actual resource limiting tests are within the `codejail` package itself.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Make a copy of codejail settings just for this test class.
+        # Set a global REALTIME limit of 100.
+        # Set a REALTIME limit override of 200 for a special course.
+        cls.test_codejail_settings = (getattr(settings, 'CODE_JAIL', None) or {}).copy()
+        cls.test_codejail_settings['limits'] = {
+            'REALTIME': 100,
+        }
+        cls.test_codejail_settings['limit_overrides'] = {
+            'course-v1:my+special+course': {'REALTIME': 200, 'NPROC': 30},
+        }
+        cls.configure_codejail(cls.test_codejail_settings)
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+
+        # Re-apply original configuration.
+        cls.configure_codejail(getattr(settings, 'CODE_JAIL', None) or {})
+
+    @staticmethod
+    def configure_codejail(codejail_settings):
+        """
+        Given a `settings.CODE_JAIL` dictionary, apply it to the codejail package.
+
+        We use the `ConfigureCodeJailMiddleware` that comes with codejail.
+        """
+        with override_settings(CODE_JAIL=codejail_settings):
+            # To apply `settings.CODE_JAIL`, we just intialize an instance of the
+            # middleware class. We expect it to apply to changes, and then raise
+            # "MiddlewareNotUsed" to indicate that its work is done.
+            # This is exactly how the settings are applied in production (except the
+            # middleware is automatically initialized because it's an element of
+            # `settings.MIDDLEWARE`).
+            try:
+                ConfigureCodeJailMiddleware()
```

Comment:
How is this configuration initialized when it's being executed as a celery task, where there is no middleware active?

Reply:
@ormsbee Oh, great question. I have no idea. I'll look into that.

Reply:
For onlookers: As of right now, it is still an open question whether this middleware is initialized on Celery workers and, if so, how it is initialized.

At worst, this PR will be a no-op for async codejail tasks, so I'm merging anyway. But this is still being investigated as part of https://openedx.atlassian.net/browse/CR-2862.